sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureNamedKeyCredential.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Represents a credential with a key name and the key and uses the key to authenticate to an Azure Service.
+ */
+public final class AzureNamedKeyCredential {
+    private final ClientLogger logger = new ClientLogger(AzureNamedKeyCredential.class);
+
+    private volatile AzureNamedKey credentials;
+
+    /**
+     * Creates a credential with specified {@code name} that authorizes request with the given {@code key}.
+     *
+     * @param name The name of the key credential.
+     * @param key The key used to authorize requests.
+     * @throws NullPointerException If {@code key} or {@code name} is {@code null}.
+     * @throws IllegalArgumentException If {@code key} or {@code name} is an empty string.
+     */
+    public AzureNamedKeyCredential(String name, String key) {
+        validateInputParameters(name, key);
+        this.credentials = new AzureNamedKey(name, key);
+    }
+
+    /**
+     * Retrieves the {@link AzureNamedKey} containing the name and key associated with this credential.
+     *
+     * @return The {@link AzureNamedKey} containing the name and key .
+     */
+    public AzureNamedKey getAzureNamedKey() {
```

Comment:
This is returning non-public API in a public API

Reply:
AzureNamedKey is a public class.